pkg/internal/controller/controller.go

```diff
@@ -58,12 +57,18 @@ type Controller struct {
 	// the Queue for processing
 	Queue workqueue.RateLimitingInterface
 
+	// startedSources maintains a list of sources that have already started.
+	startedSources []source.Source
```

Comment:
Just curious, would it be possible to not stop the controller but selectively remove watches for certain GVKs? That was been one of the common asks by the operator authors.

Reply:
It should be possible to keep a reference to the source before watching it in your controller, you can then call the `Stop` function on this source reference. I'm not sure if there is a better/ more elegant way to do this.

Reply:
Thanks @inteon. I'll test it out!